### DISCUSSION TITLE
How to disable read-only mode?

### DISCUSSION THREAD

**Discussion by kfconeone** [elsa-workflows/elsa-studio#330] [Q&A]
![image](https://github.com/user-attachments/assets/8f8cf0e1-9256-4ce3-a84d-dd9b7074fc4d)

I am new to Elsa, and yesterday I just skimmed the documents and create a server and a studio to try something, as above it said it's read-only mode, how to turn it off?

can't find any clue in the documents...

**Comment by kfconeone** [comment]
Ok... After digging into some source code, I found some clues. I suspect that the read-only mode issue is highly likely caused by version inconsistencies between the packages in my Elsa Server and Studio.

The root cause was that I switched the persistent storage to PostgreSQL, which triggered a series of version downgrades. After I unified all versions to 3.1.0, the problem was resolved.

below are what I changed
- server
```xml
<PackageReference Include="Elsa" Version="3.1.0" />
    <PackageReference Include="Elsa.CSharp" Version="3.1.0" />
    <PackageReference Include="Elsa.EntityFrameworkCore" Version="3.1.0" />
    <PackageReference Include="Elsa.EntityFrameworkCore.PostgreSql" Version="3.1.0" />
    <PackageReference Include="Elsa.Identity" Version="3.1.0" />
    <PackageReference Include="Elsa.Scheduling" Version="3.1.0" />
    <PackageReference Include="Elsa.Workflows.Api" Version="3.1.0" />
```

- studio
```xml
    <PackageReference Include="Elsa.Api.Client" Version="3.1.0" />
    <PackageReference Include="Elsa.Studio" Version="3.1.0" />
    <PackageReference Include="Elsa.Studio.Core.BlazorWasm" Version="3.1.0" />
    <PackageReference Include="Elsa.Studio.Login.BlazorWasm" Version="3.1.0" />
```

**Reply by Nekrosis1** [reply]
Tip: It is possible to have minor version differences between Elsa Studio and Elsa Core, however, the Elsa.Api.Client (on the client) needs to be the same version as the packages on the Server.
